src/models.py

```diff
@@ -0,0 +1,45 @@
+from passlib import hash
+from sqlalchemy import Column, ForeignKey, Integer, String, orm, types
+
+import database
+from schemas import StatusEnum
+
+
+class User(database.Base):
+    __tablename__ = "users"
+    id = Column(Integer, primary_key=True, index=True)
+    email = Column(String, unique=True, index=True)
+    hashed_password = Column(String)
+
+    tasks = orm.relationship("Task", back_populates="user")
+
+    def verify_password(self, password: str):
+        return hash.bcrypt.verify(password, self.hashed_password)
+
+
+class ChoiceType(types.TypeDecorator):
+    impl = types.String
+
+    def __init__(self, choices, **kw):
+        self.choices = dict(choices)
+        super(ChoiceType, self).__init__(**kw)
+
+    def process_bind_param(self, value, dialect):
```

Comment:
`dialect` does not look like used here.

Reply:
Yes, I agree. But the method `process_bind_param` is from `sqlalchemy. types` with these params and `NotImplementedError` so IDE highlights(warning) it without the `dialect` param.